### ISSUE TITLE
PPOAgent hyperparameters 

### ISSUE THREAD

**Issue by zanellar** [tensorforce/tensorforce#535]
Hello,
Where can I find some documentation about the **PPOAgent setting** (and possibly also of the others agents)? I see there are **many parameters available** (batching_capacity, baseline, summarizer, actions_exploration, variable_noise,  etc.) **but I can't understand how to set them**. For example how can I change the exploration or set the baseline or decide when to execute the training?

Thank you so much!

**Comment by AlexKuhnle** [comment]
Hi, unfortunately there is not much more documentation than the one in the code/readthedocs. If you haven't seen it yet, [this issue](https://github.com/reinforceio/tensorforce/issues/485) definitely will be of interest to you. Example configs [here](https://github.com/reinforceio/tensorforce/tree/master/examples/configs) may also help: `ppo.json` contains a baseline, `dqn_ue4.json` uses exploration, for instance. Others can also be found specified there. `batching_capacity` is more a performance parameter and its effect on learning is negligible.

I hope I soon manage to improve the documentation in general. In the meantime, feel free to ask here or in Gitter about details (but note that explaining how to optimally choose parameters is difficult, as there is currently little general guidelines).

**Comment by zanellar** [comment]
OK, thank you! I'll try to understand it on my own for now. 
Anyway, maybe you can answer this for me: 
by setting `batched_observe=True` and `batching_capacity=1000`, the training is performed every 1000 steps, correct? What happen instead if I set `batched_observe=False` and I add `terminal=True` in the observe method when the episode ends, like `agent.observe(reward=reward, terminal=done)`? Is the training done only when the episode is over? If not, how can do it?

Thank you again

**Comment by AlexKuhnle** [comment]
Sure: `batched_observe` and `batching_capacity` do not (really) change the update logic, they just specify (if turned on) that internal `observe()` is not called at every `agent.observe()` to reduce the number of "expensive" TensorFlow session calls. `batching_capacity` sets how many timesteps should be buffered at most before doing the internal call (episode terminal also forces this call). Internally, there are other buffers storing the batch of recent observations and handling the update logic as specified by `update_mode`. Training based on episodes, so at the end of an episode, can be done via `unit='episodes'`, see e.g. [here](https://github.com/reinforceio/tensorforce/blob/master/examples/configs/ppo.json).

**Comment by AlexKuhnle** [comment]
Improved
